src/lib/axes/axis_utils.test.ts

```diff
@@ -205,36 +205,90 @@ describe('Axis computational utils', () => {
-    const expectedAxisPositions = [
-      { label: '0', position: 100, value: 0 },
-      { label: '0.1', position: 90, value: 0.1 },
-      { label: '0.2', position: 80, value: 0.2 },
-      { label: '0.3', position: 70, value: 0.3 },
-      { label: '0.4', position: 60, value: 0.4 },
-      { label: '0.5', position: 50, value: 0.5 },
-      { label: '0.6', position: 40, value: 0.6 },
-      { label: '0.7', position: 30, value: 0.7 },
-      { label: '0.8', position: 20, value: 0.8 },
-      { label: '0.9', position: 10, value: 0.9 },
-      { label: '1', position: 0, value: 1 },
-    ];
-    expect(axisPositions).toEqual(expectedAxisPositions);
-
-    // histogram mode axis ticks should add an additional tick
-    const xBandDomain: XDomain = {
-      type: 'xDomain',
-      scaleType: ScaleType.Linear,
-      domain: [0, 100],
-      isBandScale: true,
-      minInterval: 10,
-    };
-    const xScale = getScaleForAxisSpec(horizontalAxisSpec, xBandDomain, [yDomain], 1, 0, 100, 0);
-    const histogramAxisPositions = getAvailableTicks(horizontalAxisSpec, xScale!, 1, true);
-    const histogramTickLabels = histogramAxisPositions.map((tick: AxisTick) => tick.label);
-    expect(histogramTickLabels).toEqual(['0', '10', '20', '30', '40', '50', '60', '70', '80', '90', '100', '110']);
+  describe('getAvailableTicks', () => {
+    test('should compute to end of domain when histogram mode not enabled', () => {
+      const enableHistogramMode = false;
+      const scale = getScaleForAxisSpec(verticalAxisSpec, xDomain, [yDomain], 0, 0, 100, 0);
+      const axisPositions = getAvailableTicks(verticalAxisSpec, scale!, 0, enableHistogramMode);
+      const expectedAxisPositions = [
+        { label: '0', position: 100, value: 0 },
+        { label: '0.1', position: 90, value: 0.1 },
+        { label: '0.2', position: 80, value: 0.2 },
+        { label: '0.3', position: 70, value: 0.3 },
+        { label: '0.4', position: 60, value: 0.4 },
+        { label: '0.5', position: 50, value: 0.5 },
+        { label: '0.6', position: 40, value: 0.6 },
+        { label: '0.7', position: 30, value: 0.7 },
+        { label: '0.8', position: 20, value: 0.8 },
+        { label: '0.9', position: 10, value: 0.9 },
+        { label: '1', position: 0, value: 1 },
+      ];
+      expect(axisPositions).toEqual(expectedAxisPositions);
+    });
+
+    test('should extend ticks to domain + minInterval in histogram mode for linear scale', () => {
+      const enableHistogramMode = true;
+      const xBandDomain: XDomain = {
+        type: 'xDomain',
+        scaleType: ScaleType.Linear,
+        domain: [0, 100],
+        isBandScale: true,
+        minInterval: 10,
+      };
+      const xScale = getScaleForAxisSpec(horizontalAxisSpec, xBandDomain, [yDomain], 1, 0, 100, 0);
+      const histogramAxisPositions = getAvailableTicks(horizontalAxisSpec, xScale!, 1, enableHistogramMode);
+      const histogramTickLabels = histogramAxisPositions.map((tick: AxisTick) => tick.label);
```

Comment:
I think giving the type for HOF is a little redundant given the array should have the type, also destructuring. Just a thought, no big deal.

```suggestion
      const histogramTickLabels = histogramAxisPositions.map(({ label }) => label);
```

Reply:
I added the destructuring but kept the type annotation https://github.com/elastic/elastic-charts/pull/251/commits/5d338111bddbde46a38178c670f3b1d06912e350

This is maybe something we will want to discuss more as a whole team, but in these kinds of cases, I prefer to see the type (even though it is not strictly necessary) because it helps me directly see the type of thing I'm working with without having to reference the array being mapped over; the explicitness isn't just for checking purposes but also reasoning purposes.  Yes, it takes like milliseconds to check the type of the array and then infer the type of the function parameter, but making the type explicit frees up my working memory so that I don't have to do that inference myself.  It's a small marginal benefit to be sure, but I'm not sure there's any disadvantage to having the type annotation here, so I'd take the small marginal benefit when I can.

Maybe this is something we can discuss in one of our upcoming syncs with @markov00?